package/cloudshell/cm/customscript/domain/github_script_downloader.py

```diff
@@ -0,0 +1,226 @@
+        """
+        :type logger: Logger
+        :type allowed_filename_pattern: str
+        :type cancel_sampler: CancellationSampler
+        """
+        self.logger = logger
+        self.filename_pattern = allowed_filename_pattern
+        self.cancel_sampler = cancel_sampler
+
+    def download(self, url, auth):
+        """
+        :type url: str
+        :type auth: HttpAuth
+        :rtype: ScriptsData
+        """
+        self._validate_github_url(url)
+        url_data = self._extract_data_from_url(url)
+
+        if self.is_url_to_file(url_data.path):
+            self.logger.info('We got a URL to a file directly,'
+                             ' will download single file')
+            self.logger.info('api_dl_url='+url_data.api_dl_url)
+            file_name, file_txt = self._download_single_file(auth, url_data.api_dl_url)
+            self.logger.info('Done')
+            return ScriptsData(main_script=ScriptFile(file_name, file_txt))
+        else:
+            self.logger.info('URL is not for a direct file, dir option (multiple files) is not currently supported')
+        '''
+        else:
+            # download all files in folder
+            self.logger.info('URL is not for a direct file, will try to treat as a dir')
+
+            # 1. list files in repo
+            files_list_in_path = self._get_list_of_files_in_path(auth, url_data)
+
+            # 2. filter out junk files from list
+            files_list_in_path = self._remove_junk_files_and_folders_from_files_list(files_list_in_path)
+
+            # 3. get & verify main.ps1/sh/bash exists
+            main_file_info = self.get_main_file(files_list_in_path)
+
+            # 4. download main file
+            main_file_url = self._build_url_to_download_single_file(url_data, main_file_info)
+            self.logger.info('Downloading main file {} from {}'.format(main_file_info['name'], main_file_url))
+            main_file_name, main_file_txt = self._download_single_file(auth, main_file_url)
+            self.logger.info('Done downloading main file')
+
+            # 5. download additional files
+            # todo: download in parallel
+            additional_files = []
+            for file_info in files_list_in_path:
+                # ignore main file
+                if file_info['name'] == main_file_info['name']:
+                    continue
+
+                file_url = self._build_url_to_download_single_file(url_data, file_info)
+                self.logger.info('Downloading single file {} from {}'.format(file_info['name'], file_url))
+                file_name, file_txt = self._download_single_file(auth, file_url)
+                additional_files.append(ScriptFile(name=file_name, text=file_txt))
+                self.logger.info('Done downloading file {}'.format(file_name))
+
+            return ScriptsData(ScriptFile(main_file_name, main_file_txt), additional_files)
+        '''
+
+    def _get_list_of_files_in_path(self, auth, url_data):
```

Comment:
It looks like garbage from a copy paste. Please clean up all unused methods.

Reply:
done